src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -384,6 +377,19 @@ RouterExecutorRun(QueryDesc *queryDesc, ScanDirection direction, long count)
 		(*destination->rStartup)(destination, operation, queryDesc->tupDesc);
 	}
 
+	/* we only support returning nothing or scanning forward */
```

Comment:
Would anything bad happen if we just set `routerState->eof_underlying` to `false` inside of the `NoMovement` block?